### DISCUSSION TITLE
Multi-relational Link prediction on a knowledge graph with homogeneous features

### DISCUSSION THREAD

**Discussion by LuisaWerner** [pyg-team/pytorch_geometric#8021] [Q&A]
Hi all, 

I want to do link prediction on the [Wikialumni dataset](https://arxiv.org/abs/2106.07297) which is a homogeneous multirelational graph. That means, the nodes features are of the same dimension while the edges have different types. 
My goal is to predict not only whether a link between two nodes exists but also which type of link exists between them. I want to do this in a multi-label setting, namely that several links of different types may exist between two nodes. 

Is there any example doing something similar? 

I already had a look at a [link prediction tutorial with RGCN](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/rgcn_link_pred.py). But there, only the links are predicted and not their types. The types are only used during message passing. 

I also found the [thread on link prediction on heterogeneous graphs](https://github.com/pyg-team/pytorch_geometric/issues/3958) opened by @sophiakrix, but it looks different to me since the dataset has heterogeneous node types and only one link between two nodes can exist. (Or did I get this wrong @sophiakrix and you predicted multiple links? If yes, I would be very happy to have more details on how you did it)

### To precise my task a bit more: 
The dataset is a `Data` object: ```{Data} (x=[52683, 300], edge_index=[2, 121842], y=[52683], num_classes=29, edge_type=[121842])```
I want to predict the edges in ```edge_index``` and their type in ```edge_type```

To split the links in train, valid and test, I use ```RandomLinkSplit``` and rename the edge_type to edge_label so that the types are kept in the splits as labels. 
I think it would be required to split the edges not only randomly but keep the types balanced. Is there a possibility to do it with ```RandomLinkSplit``` on a ```Data``` object? (Feature request :-) )

```
data.edge_label = data.edge_type
transform = RandomLinkSplit(is_undirected=False,
                                    num_val=0.1,
                                    num_test=0.3,
                                    add_negative_train_samples=False,
                                    neg_sampling_ratio=0.0)
train_data, val_data, test_data = transform(data)
```

As a model I think of using an encoder-decoder model such as [Variational Graph AutoEncoder](https://arxiv.org/abs/1611.07308) that returns in the Decoder a 3D tensor that predicts not only the link but also its type or predict the different types sequentially to avoid CUDA out of memory. 

Is anyone working on a similar task? Or has some ideas how batching can work in this setting to avoid memory issues? 

Any help would be appreciated!

**Comment by Gaurang111** [comment]
@LuisaWerner were you able to find any way for Multi-category link prediction? I am working on a similar kinda problem, it might help me.
